packages/contracts/src/MultisigSetup.sol

```diff
@@ -79,19 +71,32 @@ contract MultisigSetup is PluginUpgradeableSetup {
     }
 
     /// @inheritdoc IPluginSetup
-    /// @dev Nothing needs to happen for the update.
+    /// @dev Revoke the upgrade plugin permission to the DAO for all builds prior the current one (3).
     function prepareUpdate(
         address _dao,
-        uint16 _currentBuild,
+        uint16 _fromBuild,
         SetupPayload calldata _payload
     )
         external
-        pure
+        view
         override
         returns (bytes memory initData, PreparedSetupData memory preparedSetupData)
-    // solhint-disable-next-line no-empty-blocks
     {
-
+        (initData);
+        if (_fromBuild < 3) {
+            PermissionLib.MultiTargetPermission[]
+                memory permissions = new PermissionLib.MultiTargetPermission[](1);
+
+            permissions[0] = PermissionLib.MultiTargetPermission({
+                operation: PermissionLib.Operation.Revoke,
+                where: _payload.plugin,
+                who: _dao,
+                condition: PermissionLib.NO_CONDITION,
+                permissionId: Multisig(IMPLEMENTATION).UPGRADE_PLUGIN_PERMISSION_ID()
```

Comment:
Just a remark: we want to have libraries storing these commonly needed constants instead of redefining them for each plugin. See https://aragonassociation.atlassian.net/browse/APP-1172.

Generally, we want to put more things into libraries. See https://aragonassociation.atlassian.net/browse/OS-953